src/context.hpp

```diff
@@ -73,11 +99,27 @@ struct cvk_context : public _cl_context,
         return size <= m_device->max_mem_alloc_size();
     }
 
+    int get_property_index(const int prop) {
+        for (unsigned i = 0; i < m_properties.size(); i += 2) {
+            if (m_properties[i] == prop) {
+                return i + 1;
+            }
+        }
+        return -1;
+    }
+
+    size_t get_printf_buffersize() { return m_printf_buffersize; }
+    cvk_printf_callback_t get_printf_callback() { return m_printf_callback; }
+    void* get_printf_userdata() { return m_printf_userdata; }
+
 private:
     cvk_device* m_device;
     std::mutex m_callbacks_lock;
     std::vector<cvk_context_callback> m_destuctor_callbacks;
     std::vector<cl_context_properties> m_properties;
+    size_t m_printf_buffersize;
+    cvk_printf_callback_t m_printf_callback;
+    void* m_printf_userdata;
```

Comment:
```suggestion
    void* m_userdata;
```
And change all the places where it is used.